app/models/user.rb

```diff
@@ -24,6 +24,10 @@ class User < ApplicationRecord
   # TODO: samuel - ActiveStorage validations needs to be discussed and implemented.
   validate :avatar_validation
 
+  validates :lang, inclusion: { in: I18n.available_locales.map(&:to_s) }
+
+  before_validation :ensure_lang_set, on: :create
```

Comment:
I'm not so sure about this - there's 3 cases that we need to cover:
- If `DEFAULT_LOCALE` is set, then the user should be created with `DEFAULT_LOCALE`
- If `DEFAULT_LOCALE` is not set, then the user should be created with the same language as their browser
- If the user manually changes their language, then that should be the value of language